Subject: [PATCH] removed not existing anymore ctr and drgb modules
---
Index: funtoo_ramdisk/support/module_configs/full/modules.copy
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/funtoo_ramdisk/support/module_configs/full/modules.copy b/funtoo_ramdisk/support/module_configs/full/modules.copy
--- a/funtoo_ramdisk/support/module_configs/full/modules.copy	(revision dec2124ac37c082e71141bfd5522917553bc6792)
+++ b/funtoo_ramdisk/support/module_configs/full/modules.copy	(date 1725656585664)
@@ -68,6 +68,4 @@
 cryptd
 crypto_null
 crypto_simd
-ctr
-drbg
 xts
\ No newline at end of file
Index: funtoo_ramdisk/support/module_configs/full/modules.autoload
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/funtoo_ramdisk/support/module_configs/full/modules.autoload b/funtoo_ramdisk/support/module_configs/full/modules.autoload
--- a/funtoo_ramdisk/support/module_configs/full/modules.autoload	(revision dec2124ac37c082e71141bfd5522917553bc6792)
+++ b/funtoo_ramdisk/support/module_configs/full/modules.autoload	(date 1725656585684)
@@ -88,6 +88,4 @@
 cryptd
 crypto_null
 crypto_simd
-ctr
-drbg
 xts
\ No newline at end of file
Index: funtoo_ramdisk/support/module_configs/zfs/modules.copy
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/funtoo_ramdisk/support/module_configs/zfs/modules.copy b/funtoo_ramdisk/support/module_configs/zfs/modules.copy
--- a/funtoo_ramdisk/support/module_configs/zfs/modules.copy	(revision dec2124ac37c082e71141bfd5522917553bc6792)
+++ b/funtoo_ramdisk/support/module_configs/zfs/modules.copy	(date 1725656585696)
@@ -68,8 +68,6 @@
 cryptd
 crypto_null
 crypto_simd
-ctr
-drbg
 xts
 
 [zfs]
Index: funtoo_ramdisk/support/module_configs/zfs/modules.autoload
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/funtoo_ramdisk/support/module_configs/zfs/modules.autoload b/funtoo_ramdisk/support/module_configs/zfs/modules.autoload
--- a/funtoo_ramdisk/support/module_configs/zfs/modules.autoload	(revision dec2124ac37c082e71141bfd5522917553bc6792)
+++ b/funtoo_ramdisk/support/module_configs/zfs/modules.autoload	(date 1725656585704)
@@ -88,8 +88,6 @@
 cryptd
 crypto_null
 crypto_simd
-ctr
-drbg
 xts
 
 [zfs]
