Subject: [PATCH] removed not existing anymore linear and multipath modules
---
Index: funtoo_ramdisk/support/module_configs/zfs/modules.autoload
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/funtoo_ramdisk/support/module_configs/zfs/modules.autoload b/funtoo_ramdisk/support/module_configs/zfs/modules.autoload
--- a/funtoo_ramdisk/support/module_configs/zfs/modules.autoload	(revision 48b20e15f647aeba0d38ccb4b8e4d7a8d8fd55c2)
+++ b/funtoo_ramdisk/support/module_configs/zfs/modules.autoload	(date 1723388458041)
@@ -37,8 +37,6 @@
 raid1
 raid456
 raid10
-linear
-multipath
 
 [ata]
 kernel/drivers/ata/**
Index: funtoo_ramdisk/support/module_configs/full/modules.autoload
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/funtoo_ramdisk/support/module_configs/full/modules.autoload b/funtoo_ramdisk/support/module_configs/full/modules.autoload
--- a/funtoo_ramdisk/support/module_configs/full/modules.autoload	(revision 48b20e15f647aeba0d38ccb4b8e4d7a8d8fd55c2)
+++ b/funtoo_ramdisk/support/module_configs/full/modules.autoload	(date 1723388694330)
@@ -37,8 +37,6 @@
 raid1
 raid456
 raid10
-linear
-multipath
 
 [ata]
 kernel/drivers/ata/**
